src/core/public/workspace/workspaces_service.ts

```diff
@@ -32,23 +51,45 @@ export class WorkspaceService implements CoreService<WorkspaceSetup, WorkspaceSt
   private workspaceList$ = new BehaviorSubject<WorkspaceAttribute[]>([]);
   private currentWorkspace$ = new BehaviorSubject<WorkspaceAttribute | null>(null);
   private workspaceEnabled$ = new BehaviorSubject<boolean>(false);
+  private _renderWorkspaceMenu: WorkspaceMenuRenderFn | null = null;
 
   public setup(): WorkspaceSetup {
     return {
       currentWorkspaceId$: this.currentWorkspaceId$,
       currentWorkspace$: this.currentWorkspace$,
       workspaceList$: this.workspaceList$,
       workspaceEnabled$: this.workspaceEnabled$,
+      registerWorkspaceMenuRender: (render: WorkspaceMenuRenderFn) =>
+        (this._renderWorkspaceMenu = render),
     };
   }
 
-  public start(): WorkspaceStart {
-    return {
+  public start({
+    http,
+    application,
+  }: {
+    application: InternalApplicationStart;
+    http: HttpSetup;
+  }): WorkspaceStart {
+    const observables = {
       currentWorkspaceId$: this.currentWorkspaceId$,
       currentWorkspace$: this.currentWorkspace$,
       workspaceList$: this.workspaceList$,
       workspaceEnabled$: this.workspaceEnabled$,
     };
+    return {
+      ...observables,
+      renderWorkspaceMenu: () => {
+        if (this._renderWorkspaceMenu) {
+          return this._renderWorkspaceMenu({
+            basePath: http.basePath,
+            getUrlForApp: application.getUrlForApp,
+            observables,
+          });
+        }
+        return null;
+      },
+    };
   }
 
   public async stop() {
```

Comment:
Please also remember to unsubscribe workspaceEnabled$

Reply:
The code refactor is not supposed to add the missing `unsubscribe`, so I didn't notice it. But I can add it in this PR.

Reply:
Do we need to set `_renderWorkspaceMenu` to null after stop?

Reply:
Not necessary, but I can add that, better than not.